Objects/typevarobject.c

```diff
@@ -458,6 +461,75 @@ unpack_typevartuples(PyObject *params)
     }
 }
 
+static PyObject *
+typeparam_reduce_anonymous(PyObject *self, PyObject *owner)
+{
+    assert(PyTuple_Check(owner) && PyTuple_GET_SIZE(owner) == 3);
+    PyObject *module_name = PyTuple_GET_ITEM(owner, 0);
+    PyObject *qualname = PyTuple_GET_ITEM(owner, 1);
+    PyObject *index = PyTuple_GET_ITEM(owner, 2);
+    assert(PyUnicode_Check(module_name));
+    assert(PyUnicode_Check(qualname));
+    assert(PyLong_Check(index));
+    PyObject *ret = NULL;
+    PyObject *typing = NULL;
+    PyObject *restore_func = NULL;
+    PyObject *module = NULL;
+    PyObject *path = NULL;
+    PyObject *obj = NULL;
+    PyObject *args = NULL;
+
+    typing = PyImport_ImportModule("typing");
+    if (typing == NULL) {
+        goto done;
+    }
+    restore_func = PyObject_GetAttrString(typing, "_restore_anonymous_typeparam");
```

Comment:
Out of curiosity, but does this one really need to import `typing`? can't we do it in C? The function doesn't seem to complex right?

Reply:
I think the trouble is that we need to pass the function object as part of the `__reduce__` return value. Maybe we can expose it in `_typing` and return a reference directly to the function object though.

Reply:
Ok, moved into `_typing`. But not sure how much of a simplification it is given that `_typing` itself needs to be imported for this and that `typing` is imported anyway even if you just do a `class cls[T]: pass`.

Reply:
> that typing is imported anyway even if you just do a class cls[T]: pass.

Oh, I wasn't aware of this. Then it's probably better to keep something simple. You could `_Py_ID()` to cache `_restore_anonymous_typeparam`. How frequent would this code be called anyway?

Reply:
> How frequent would this code be called anyway?

Pickling or unpickling instances of generic classes mostly, maybe copy/deepcopy using `__reduce__`. Not sure `_Py_ID` would have any effect at all given everything else that is happens.

Reply:
Considering this, I think we can leave it as is. Maybe if performances are impacted significantly, we'll rework it but for now it's fine.

Reply:
Just to clarify, I understand "leave as is" means as is now with `_typing._restore_anonymous_typeparam`, and not put it back in `typing`.